packages/vsce/src/SshConfigUtils.ts

```diff
@@ -34,145 +39,262 @@ export class SshConfigUtils {
     public static async promptForProfile(profileName?: string): Promise<imperative.IProfileLoaded | undefined> {
         const zoweExplorerApi = ZoweVsCodeExtension.getZoweExplorerApi();
         const profCache = zoweExplorerApi.getExplorerExtenderApi().getProfilesCache();
-        const profInfo = await zoweExplorerApi.getExplorerExtenderApi().getProfilesCache().getProfileInfo();
-        const configExists = profInfo.getTeamConfig().exists;
+        const profInfo = await profCache.getProfileInfo();
 
-        if (profileName != null) {
+        if (profileName) {
             return profCache.getLoadedProfConfig(profileName, "ssh");
         }
 
         const sshProfiles = (await profCache.fetchAllProfilesByType("ssh")).filter(
-            ({ name, profile }) => name != null && profile?.host,
+            ({ name, profile }) => name && profile?.host,
         );
 
-        // get configs via migrateSshConfig() and remove options that have host name that already exists as an ssh profile on the config file
-        const migratedConfigs = (await ZClientUtils.migrateSshConfig()).filter(
+        // Get configs from ~/.ssh/config
+        const migratedConfigs = await ZClientUtils.migrateSshConfig();
+
+        // Parse to remove migratable configs that already exist on the team config
+        const filteredMigratedConfigs = migratedConfigs.filter(
             (migratedConfig) => !sshProfiles.some((sshProfile) => sshProfile.profile?.host === migratedConfig.hostname),
         );
 
+        // Choose between adding a new ssh host, an existing team config profile, and migrating from config.
         const qpItems: vscode.QuickPickItem[] = [
             { label: "$(plus) Add New SSH Host..." },
-            ...sshProfiles.map((prof) => ({ label: prof.name!, description: prof.profile?.host })),
-            { label: "Migrate From SSH Config", kind: vscode.QuickPickItemKind.Separator },
-            ...migratedConfigs.map((config) => ({ label: config.name!, description: config.hostname })),
+            ...sshProfiles.map(({ name, profile }) => ({
+                label: name!,
+                description: profile?.host,
```

Comment:
Again, not a huge fan of bypassing TS compile-time errors 😅 
I know we are already filtering the list of ssh profiles above, so, sometimes it makes sense to use it 😅 
```
        const sshProfiles = (await profCache.fetchAllProfilesByType("ssh")).filter(
            ({ name, profile }) => name && profile?.host,
        );
```

In theory we could also do this... 😅 
```suggestion
                label: name!,
                description: profile!.host!,
```

But again, not a big fan (personally) 😓